From 9a81ac06254eb8e942b0d155fa19635749e5a0b5 Mon Sep 17 00:00:00 2001
From: David Davis <daviddavis@users.noreply.github.com>
Date: Thu, 17 Nov 2022 10:11:20 -0500
Subject: [PATCH] Rewind repo.xml.key file before hashing.

---
 pulp_rpm/app/tasks/publishing.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pulp_rpm/app/tasks/publishing.py b/pulp_rpm/app/tasks/publishing.py
index de06eb35..e1b9becf 100644
--- a/pulp_rpm/app/tasks/publishing.py
+++ b/pulp_rpm/app/tasks/publishing.py
@@ -683,6 +683,7 @@ def generate_repo_metadata(
         with open(pubkey_name, "wb+") as f:
             f.write(signing_service.public_key.encode("utf-8"))
             f.flush()
+            f.seek(0)
             PublishedMetadata.create_from_file(
                 relative_path=os.path.join(repodata_path, pubkey_name),
                 publication=publication,
